### PR TITLE
Hardcode JS library version in `checkVersion`

### PR DESCRIPTION
**Pull request by jwajgelt**
## Summary
We hardcode the version of Reanimated here in order to compare it with the version used to build the native part of the library in runtime.

Ideally, we'd grab this version from the `package.json` file instead, but this causes issues when packaging the library due to misaligned relative paths.

### CODE REVIEW

File: src/reanimated2/platform-specific/checkVersion.ts [hunk 1]
```diff
@@ -1,4 +1,9 @@
-import { version as jsVersion } from '../../../package.json';
+/**
+ * We hardcode the version of Reanimated here in order to compare it
+ * with the version used to build the native part of the library in runtime.
+ * Remember to keep this in sync with the version declared in `package.json`
```
Comment (piaskowyk) [inline src/reanimated2/platform-specific/checkVersion.ts:4]: `Remember to keep this in sync with the version declared in package.json` - This is the tricky part 😁
By my experience this is not a good practices to keep something variable hardcoded in code.

Comment (piaskowyk) [reply src/reanimated2/platform-specific/checkVersion.ts:4]: I would rather prefer to generate file contains Reaniated version (version.json) during package build

Comment (piaskowyk) [reply src/reanimated2/platform-specific/checkVersion.ts:4]: Or we can keep it in code but we need to add CI to test it

Comment (piaskowyk) [reply src/reanimated2/platform-specific/checkVersion.ts:4]: I like the last idea 😎

Comment (jwajgelt) [reply src/reanimated2/platform-specific/checkVersion.ts:4]: I know, but we've tried a ton of different things, and they just refuse to work :(
We'll put a check in the CI that they're in sync, and it should be fine.

**Review by piaskowyk — APPROVED** [review]

**Pull request merged**